### PR TITLE
Tickets/damd 37

### PR DESCRIPTION
**Pull request by pychabau**
We have added the "redshift candidates" part to datamodel.txt which describes the 1d DRP pipeline output datamodel for redshift/lines measurement

### CODE REVIEW

File: datamodel.txt [hunk 1]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+The path would be
+   tract/patch/pfsZcandidate-*.fits
+
+The file will have several HDUs:
+
+HDU#0 PDU
+
+HDU#1 LAMBDA_SCALE              Binary table            [FITS BINARY TABLE]     NROW
+        Columns for :
+        WAVELENGTH              FLOAT                   Wavelength scale
```
Comment (PaulPrice) [inline datamodel.txt]: What's the purpose of this table? What does this array of wavelength link up with?
Units should be nm.

File: datamodel.txt [hunk 2]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+
+HDU#2 ZCANDIDATES               Binary table            [FITS BINARY TABLE]     NCANDIDATE
+        Colums for :
+        Z                       FLOAT                   Redshift
+        Z_ERR                   FLOAT                   Redshift error (negative for invalid fit)
+        ZRANK                   INT32                   Rank of redshift candidate ( best = 1 )
+        RELIABILITY             FLOAT                   Reliability of redshift
```
Comment (PaulPrice) [inline datamodel.txt]: "Reliability" seems rather a nebulous concept. Do you have a firm definition for it?

File: datamodel.txt [hunk 3]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+        CLASS                   STR                     Spectro classification: GALAXY, QSO, STAR
+        SUBCLASS                STR                     Subclassifications
+        ZFIT                    FLOAT[NROW]             Spectrum model
+
+HDU#3 ZPDF                      Binary table            [FITS BINARY TABLE]     NPIX
+        Colums for :
+        REDSHIFT                FLOAT                   Redshift scale
+        PDF                     FLOAT                   Probability Density Function of estimated redshift
```
Comment (PaulPrice) [inline datamodel.txt]: Is this table only for the best model, or is it marginalised over all models?

File: datamodel.txt [hunk 4]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+
+HDU#4 ZLINES                    Binary table            [FITS BINARY TABLE]     NLINE
```
Comment (PaulPrice) [inline datamodel.txt]: Is this table only for the best model fit, or all models?

File: datamodel.txt [hunk 5]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+
+HDU#2 ZCANDIDATES               Binary table            [FITS BINARY TABLE]     NCANDIDATE
+        Colums for :
+        Z                       FLOAT                   Redshift
+        Z_ERR                   FLOAT                   Redshift error (negative for invalid fit)
+        ZRANK                   INT32                   Rank of redshift candidate ( best = 1 )
+        RELIABILITY             FLOAT                   Reliability of redshift
+        CLASS                   STR                     Spectro classification: GALAXY, QSO, STAR
+        SUBCLASS                STR                     Subclassifications
+        ZFIT                    FLOAT[NROW]             Spectrum model
```
Comment (PaulPrice) [inline datamodel.txt]: Suggest alternate name: "MODELFLUX"?
Units should be nJy.

File: datamodel.txt [hunk 6]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+
+HDU#3 ZPDF                      Binary table            [FITS BINARY TABLE]     NPIX
+        Colums for :
+        REDSHIFT                FLOAT                   Redshift scale
+        PDF                     FLOAT                   Probability Density Function of estimated redshift
+
+HDU#4 ZLINES                    Binary table            [FITS BINARY TABLE]     NLINE
+        Columns for :
+        LINENAME                STR                     Line name
+        LINEWAVE                FLOAT                   Catalog wavelength for this line in vacuum Angstroms
```
Comment (PaulPrice) [inline datamodel.txt]: I thought nm was the PFS standard. Why are you using Angstroms here? (I prefer Angstroms myself, but nm appears to be the standard.)

File: datamodel.txt [hunk 7]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+        LINEZ                   FLOAT                   Redshift
+        LINEZ_ERR               FLOAT                   Redshift error (negative for invalid fit)
+        LINESIGMA               FLOAT                   Gaussian width in Angstrom
+        LINESIGMA_ERR           FLOAT                   Error in gaussian width (negative for invalid fit)
+        LINEVEL                 FLOAT                   Gaussian width in km/sec
+        LINEVEL_ERR             FLOAT                   Error in gaussian width (negative for invalid fit)
+        LINEFLUX                FLOAT                   Area in gaussian fit where units are (flux-units) * Ang
```
Comment (PaulPrice) [inline datamodel.txt]: Fluxes should be in nJy, so nJy.A is a bit silly. I suggest changing the units to erg/cm^2/s.

File: datamodel.txt [hunk 8]
```diff
@@ -394,6 +394,63 @@ Note that the wavelength need not be uniformly sampled.
...
+        LINEFLUX_ERR            FLOAT                   Flux error (negative for invalid fit)
+        LINEEW                  FLOAT                   Equivalent width (Angstroms)
+        LINEEW_ERR              FLOAT                   Equivalent width error (negative for invalid fit)
+        LINECONTLEVEL           FLOAT                   Continuum level at line center
```
Comment (PaulPrice) [inline datamodel.txt]: Units should be nJy.

File: datamodel.txt (hunk 1 above)
Comment (pychabau) [reply datamodel.txt]: The WAVELENGTH array links up the ZFIT arrays (or MODELFLUX as you suggested). Each model (associated to each redshift candidate) shares the same "wavelength". This hdu might be placed after ZCANDIDATES hdu for better understanding. According with other elements of the datamodel the wavelength unit is nm.

File: datamodel.txt (hunk 3 above)
Comment (pychabau) [reply datamodel.txt]: It is marginalised over all models.

File: datamodel.txt (hunk 5 above)
Comment (pychabau) [reply datamodel.txt]: Agreed.

File: datamodel.txt (hunk 2 above)
Comment (pychabau) [reply datamodel.txt]: Yes. For given redshift candidate, the reliability is the area of the PDF peak (dz=+/- 3*10-3)

File: datamodel.txt (hunk 4 above)
Comment (pychabau) [reply datamodel.txt]: This is the table for the best model.

File: datamodel.txt (hunk 6 above)
Comment (pychabau) [reply datamodel.txt]: I used sdss definition for lines table (https://data.sdss.org/datamodel/files/BOSS_SPECTRO_REDUX/RUN2D/PLATE4/RUN1D/spZline.html) but you're right, it should be nm for PFS.

File: datamodel.txt (hunk 8 above)
Comment (pychabau) [reply datamodel.txt]: Agreed.

File: datamodel.txt (hunk 7 above)
Comment (pychabau) [reply datamodel.txt]: Agreed.

**Comment by pychabau** [comment]
Paul's remarks have been taken into account.
Pull request will be merge in few minutes.

**Pull request merged**